pkgs/applications/virtualization/docker/buildx.nix

```diff
@@ -2,13 +2,13 @@
 
 buildGoModule rec {
   pname = "docker-buildx";
-  version = "0.9.1";
+  version = "0.10.4";
 
   src = fetchFromGitHub {
     owner = "docker";
     repo = "buildx";
     rev = "v${version}";
-    sha256 = "sha256-nJR+wpWa7y8Mq6WWj1ZH/FRCtar40XP2fwyl1hMgELI=";
+    sha256 = "sha256-hYEFboZa6RGiy0wr7NEWaf5KCz/O7KGXTc6c9hMmoXk=";
   };
 
```

Comment:
Not entirely sure why, but the go tests fail for me on nixos/x86_64-linux.

If you don't want to investigate that, you could just disable those tests when building for now by setting:

```suggestion
  doCheck = false;
```

```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/3gb9smvrfg0hjdm8k95lisva47zygzp7-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
@nix { "action": "setPhase", "phase": "buildPhase" }
building
Building subPackage ./bake
Building subPackage ./bake/hclparser
Building subPackage ./build
Building subPackage ./builder
Building subPackage ./cmd/buildx
Building subPackage ./commands
Building subPackage ./commands/imagetools
Building subPackage ./docs
Building subPackage ./driver
Building subPackage ./driver/bkimage
Building subPackage ./driver/docker
Building subPackage ./driver/docker-container
Building subPackage ./driver/kubernetes
Building subPackage ./driver/kubernetes/context
Building subPackage ./driver/kubernetes/execconn
Building subPackage ./driver/kubernetes/manifest
Building subPackage ./driver/kubernetes/podchooser
Building subPackage ./driver/remote
Building subPackage ./driver/remote/util
Building subPackage ./monitor
Building subPackage ./store
Building subPackage ./store/storeutil
Building subPackage ./util/buildflags
Building subPackage ./util/cobrautil
Building subPackage ./util/confutil
Building subPackage ./util/dockerutil
Building subPackage ./util/gitutil
Building subPackage ./util/imagetools
Building subPackage ./util/logutil
Building subPackage ./util/platformutil
Building subPackage ./util/progress
Building subPackage ./util/resolver
Building subPackage ./util/tracing
Building subPackage ./util/userfunc
Building subPackage ./util/waitmap
Building subPackage ./version
buildPhase completed in 30 seconds
@nix { "action": "setPhase", "phase": "checkPhase" }
running tests
--- FAIL: TestNestedInherits (0.00s)
    --- FAIL: TestNestedInherits/nested_simple (0.00s)
        bake_test.go:1026: 
            	Error Trace:	/build/source/bake/bake_test.go:1026
            	Error:      	Not equal: 
            	            	expected: map[string]*string{"bar":(*string)(0xc000359b20), "baz":(*string)(0xc000359b30), "foo":(*string)(0xc000359b40)}
            	            	actual  : map[string]*string{"SOURCE_DATE_EPOCH":(*string)(0xc0003a9c20), "bar":(*string)(0xc0003a8020), "baz":(*string)(0xc0003a8fb0), "foo":(*string)(0xc0003a8050)}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1,2 +1,3 @@
            	            	-(map[string]*string) (len=3) {
            	            	+(map[string]*string) (len=4) {
            	            	+ (string) (len=17) "SOURCE_DATE_EPOCH": (*string)((len=9) "315532800"),
            	            	  (string) (len=3) "bar": (*string)((len=3) "234"),
            	Test:       	TestNestedInherits/nested_simple
    --- FAIL: TestNestedInherits/nested_with_overrides_first (0.00s)
        bake_test.go:1026: 
            	Error Trace:	/build/source/bake/bake_test.go:1026
            	Error:      	Not equal: 
            	            	expected: map[string]*string{"bar":(*string)(0xc000359b50), "baz":(*string)(0xc000359b60), "foo":(*string)(0xc000359b70)}
            	            	actual  : map[string]*string{"SOURCE_DATE_EPOCH":(*string)(0xc0005b7fb0), "bar":(*string)(0xc0005b6410), "baz":(*string)(0xc0005b7370), "foo":(*string)(0xc0005b7f10)}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1,2 +1,3 @@
            	            	-(map[string]*string) (len=3) {
            	            	+(map[string]*string) (len=4) {
            	            	+ (string) (len=17) "SOURCE_DATE_EPOCH": (*string)((len=9) "315532800"),
            	            	  (string) (len=3) "bar": (*string)((len=3) "234"),
            	Test:       	TestNestedInherits/nested_with_overrides_first
    --- FAIL: TestNestedInherits/nested_with_overrides_last (0.00s)
        bake_test.go:1026: 
            	Error Trace:	/build/source/bake/bake_test.go:1026
            	Error:      	Not equal: 
            	            	expected: map[string]*string{"bar":(*string)(0xc000359b80), "baz":(*string)(0xc000359b90), "foo":(*string)(0xc000359ba0)}
            	            	actual  : map[string]*string{"SOURCE_DATE_EPOCH":(*string)(0xc0005e01f0), "bar":(*string)(0xc0005e0190), "baz":(*string)(0xc000441460), "foo":(*string)(0xc0005e0150)}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1,2 +1,3 @@
            	            	-(map[string]*string) (len=3) {
            	            	+(map[string]*string) (len=4) {
            	            	+ (string) (len=17) "SOURCE_DATE_EPOCH": (*string)((len=9) "315532800"),
            	            	  (string) (len=3) "bar": (*string)((len=3) "432"),
            	Test:       	TestNestedInherits/nested_with_overrides_last
--- FAIL: TestNestedInheritsWithGroup (0.00s)
    --- FAIL: TestNestedInheritsWithGroup/nested_simple (0.00s)
        bake_test.go:1098: 
            	Error Trace:	/build/source/bake/bake_test.go:1098
            	Error:      	Not equal: 
            	            	expected: map[string]*string{"BAR":(*string)(0xc0005e0420), "FOO":(*string)(0xc0005e0430)}
            	            	actual  : map[string]*string{"BAR":(*string)(0xc0005e0aa0), "FOO":(*string)(0xc0005e1110), "SOURCE_DATE_EPOCH":(*string)(0xc0001248a0)}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1,4 +1,5 @@
            	            	-(map[string]*string) (len=2) {
            	            	+(map[string]*string) (len=3) {
            	            	  (string) (len=3) "BAR": (*string)((len=3) "fuu"),
            	            	- (string) (len=3) "FOO": (*string)((len=3) "bar")
            	            	+ (string) (len=3) "FOO": (*string)((len=3) "bar"),
            	            	+ (string) (len=17) "SOURCE_DATE_EPOCH": (*string)((len=9) "315532800")
            	            	 }
            	Test:       	TestNestedInheritsWithGroup/nested_simple
    --- FAIL: TestNestedInheritsWithGroup/nested_with_overrides_first (0.00s)
        bake_test.go:1098: 
            	Error Trace:	/build/source/bake/bake_test.go:1098
            	Error:      	Not equal: 
            	            	expected: map[string]*string{"BAR":(*string)(0xc0005e0470), "FOO":(*string)(0xc0005e0480)}
            	            	actual  : map[string]*string{"BAR":(*string)(0xc0002875b0), "FOO":(*string)(0xc0002875f0), "SOURCE_DATE_EPOCH":(*string)(0xc0002876b0)}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1,4 +1,5 @@
            	            	-(map[string]*string) (len=2) {
            	            	+(map[string]*string) (len=3) {
            	            	  (string) (len=3) "BAR": (*string)((len=3) "fii"),
            	            	- (string) (len=3) "FOO": (*string)((len=5) "baaar")
            	            	+ (string) (len=3) "FOO": (*string)((len=5) "baaar"),
            	            	+ (string) (len=17) "SOURCE_DATE_EPOCH": (*string)((len=9) "315532800")
            	            	 }
            	Test:       	TestNestedInheritsWithGroup/nested_with_overrides_first
    --- FAIL: TestNestedInheritsWithGroup/nested_with_overrides_last (0.00s)
        bake_test.go:1098: 
            	Error Trace:	/build/source/bake/bake_test.go:1098
            	Error:      	Not equal: 
            	            	expected: map[string]*string{"BAR":(*string)(0xc0005e04c0), "FOO":(*string)(0xc0005e04d0)}
            	            	actual  : map[string]*string{"BAR":(*string)(0xc0002ffb60), "FOO":(*string)(0xc0002fe580), "SOURCE_DATE_EPOCH":(*string)(0xc0002ffc60)}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1,4 +1,5 @@
            	            	-(map[string]*string) (len=2) {
            	            	+(map[string]*string) (len=3) {
            	            	  (string) (len=3) "BAR": (*string)((len=3) "fii"),
            	            	- (string) (len=3) "FOO": (*string)((len=3) "bar")
            	            	+ (string) (len=3) "FOO": (*string)((len=3) "bar"),
            	            	+ (string) (len=17) "SOURCE_DATE_EPOCH": (*string)((len=9) "315532800")
            	            	 }
            	Test:       	TestNestedInheritsWithGroup/nested_with_overrides_last
--- FAIL: TestHCLNullVars (0.00s)
    bake_test.go:1324: 
        	Error Trace:	/build/source/bake/bake_test.go:1324
        	Error:      	Not equal: 
        	            	expected: map[string]*string{"bar":(*string)(0xc0005c4770)}
        	            	actual  : map[string]*string{"SOURCE_DATE_EPOCH":(*string)(0xc0005c4740), "bar":(*string)(0xc0005c4110)}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,2 +1,3 @@
        	            	-(map[string]*string) (len=1) {
        	            	+(map[string]*string) (len=2) {
        	            	+ (string) (len=17) "SOURCE_DATE_EPOCH": (*string)((len=9) "315532800"),
        	            	  (string) (len=3) "bar": (*string)((len=3) "baz")
        	Test:       	TestHCLNullVars
--- FAIL: TestJSONNullVars (0.00s)
    bake_test.go:1359: 
        	Error Trace:	/build/source/bake/bake_test.go:1359
        	Error:      	Not equal: 
        	            	expected: map[string]*string{"bar":(*string)(0xc0005c5560)}
        	            	actual  : map[string]*string{"SOURCE_DATE_EPOCH":(*string)(0xc0005c5530), "bar":(*string)(0xc0005c4f70)}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,2 +1,3 @@
        	            	-(map[string]*string) (len=1) {
        	            	+(map[string]*string) (len=2) {
        	            	+ (string) (len=17) "SOURCE_DATE_EPOCH": (*string)((len=9) "315532800"),
        	            	  (string) (len=3) "bar": (*string)((len=3) "baz")
        	Test:       	TestJSONNullVars
FAIL
FAIL	github.com/docker/buildx/bake	0.376s
FAIL
```

Reply:
Yikes, looks like those buildx tests change behavior depending on the presence of the `SOURCE_DATE_EPOCH` environment variable (we have special handling of it to work towards more reproducible docker builds).

I've opened an upstream PR: https://github.com/docker/buildx/pull/1675.

Reply:
Thanks!
I should have rephrased my original review slightly differently in hindsight :sweat_smile: 

Please change the commit message from
`docker-buildx: upgrade docker-buildx to 0.10.4`
to
`docker-buildx: 0.9.1 -> 0.10.4`

This way that commit is on par with the [past commits to that file](https://github.com/NixOS/nixpkgs/commits/082e0111a4097d0c90f635ad7bcc43474dfe3bcb/pkgs/applications/virtualization/docker/buildx.nix) and also complies with https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md :)

Reply:
thx, fixed.